src/multibase.js

```diff
@@ -53,16 +48,13 @@ function decode (bufOrString) {
     bufOrString = new Buffer(bufOrString)
   }
 
-  let decode
+  const base = getBase(code)
 
-  switch (code) {
-    case 'z': {
-      decode = (buf) => { return new Buffer(bs58.decode(buf.toString())) }
-    } break
-    default: throw errNotSupported
+  const decodeObject = {
+    base: base.getName(),
+    data: new Buffer(base.decode(bufOrString.toString()))
   }
-
-  return decode(bufOrString)
+  return decodeObject.data
```

Comment:
Looks like line 51~57 can be reduced to:

`return new Buffer(base.decode(bufOrString.toString()))`